Update FR tutorial to include file path, writer and print usage (#3058)
pytorch/tutorials: prototype_source/flight_recorder_tutorial.rst

------------------------

> There are two required environment variables to get the initial version of Flight Recorder working.

Reviewer: Nit: Can you move this setting down to **Optional settings** section?
Or alternatively fix the comment "there are two required" environment variables.
Otherwise, customers will think that "TORCH_NCCL_DEBUG_INFO_TEMP_FILE" is the required var :-)

> - ``TORCH_NCCL_DEBUG_INFO_TEMP_FILE``: Setting the path where the flight recorder will be dumped with file prefix. One file per

Reviewer: Nit: Can you move this to the **Optional settings** section?
Otherwise the comment on Line 49 can be confusing to the customer (there are two required environment variables).
Or you can fix the comment.

> - ``TORCH_NCCL_DEBUG_INFO_TEMP_FILE``: Setting the path where the flight recorder will be dumped with file prefix. The dump is one

Reviewer 2: ```suggestion
- ``TORCH_NCCL_DEBUG_INFO_TEMP_FILE``: Setting the path where the flight recorder will be dumped with file prefix. One
```

>   file per rank. The default value is ``/tmp/nccl_trace_rank_``.

Reviewer 2: ```suggestion
  file is generated per rank. The default value is ``/tmp/nccl_trace_rank_``.
```

     ``fast`` is a new experimental mode that is shown to be much faster than the traditional ``addr2line``.
     Use this setting in conjunction with ``TORCH_NCCL_TRACE_CPP_STACK`` to collect C++ traces in the Flight Recorder data.

> - If you don't want the flight recorder to be dumped into the local disk but instead onto your own storage, users can define your own writer class

Reviewer 2: ```suggestion
- If you prefer not to have the flight recorder data dumped into the local disk but rather onto your own storage, you can define your own writer class.
```

>   which inherits from class ``::c10d::DebugInfoWriter`` and then register the new writer using ``::c10d::DebugInfoWriter::registerWriter`` before

Reviewer 2: ```suggestion
This class should inherit from ``::c10d::DebugInfoWriter`` and then register the new writer using ``::c10d::DebugInfoWriter::registerWriter`` before
```

>   This class should inherit from class ``::c10d::DebugInfoWriter`` and then register the new writer using ``::c10d::DebugInfoWriter::registerWriter``

Reviewer: Suggestion: 
Can you make this a link to the code in PyTorch?

2. To run the script, use this command:

.. code:: shell

  python fr_trace.py <dump dir containing trace files> [-o <output file>]

> Or if you install PyTorch nightly build or build from scratch (with ``USE_DISTRIBUTED=1``), you can directly use the following command:

Reviewer 2: ```suggestion
If you install the PyTorch nightly build or build from scratch with ``USE_DISTRIBUTED=1``, you can use the following command directly:
```

.. code:: shell

  torchfrtrace <dump dir containing trace files> [-o <output file>]

> For now, we support two modes for the analyzer script, one is to let the script to apply some heuristics to the parsed flight recorder

Reviewer 2: ```suggestion
Currently, we support two modes for the analyzer script. The first mode allows the script to apply heuristics to the parsed flight recorder
```

> dumps to generate a report on potential culprit for the timeout/hang; the other one is to just print out raw dumps. For the latter, by

Reviewer 2: ```suggestion
dumps to generate a report identifying potential culprits for the timeout or hang. The second mode is simply outputs the raw dumps. For the latter, 
```

> default the script prints for all ranks and all ProcessGroups(PGs), and this can be narrowed down to certain ranks and PGs. Example

Reviewer 2: ```suggestion
the script, by default, prints information for all ranks and all ``ProcessGroups``(PGs). This can be narrowed down to certain ranks and PGs. An example
```

2. To run the script, use this command:

.. code:: shell

  python fr_trace.py <dump dir containing trace files> [-o <output file>]

If you install the PyTorch nightly build or build from scratch with ``USE_DISTRIBUTED=1``, you can directly use the following
command directly:

.. code:: shell

  torchfrtrace <dump dir containing trace files> [-o <output file>]

Currently, we support two modes for the analyzer script. The first mode allows the script to apply some heuristics to the parsed flight
recorder dumps to generate a report identifying potential culprits for the timeout. The second mode is simply outputs the raw dumps.
By default, the script prints flight recoder dumps for all ranks and all ``ProcessGroups``(PGs). This can be narrowed down to certain

> ranks and PGs. An example command is:

Reviewer: Suggested clarification to sentence.

can be narrowed down to certain ranks and PGs using the `selected-ranks` option.
